vunit/test/unit/test_test_suites.py

```diff
@@ -107,3 +108,38 @@ def _read_test_results(contents, expected_test_cases):
                           test_suite_name=None,
                           test_cases=expected_test_cases)
             return run._read_test_results(file_name=file_name)  # pylint: disable=protected-access
+
+    def test_exit_code(self):
+        self.assertEqual(self._test_exit_code(True), False)
+        self.assertEqual(self._test_exit_code(False), False)
+        self.assertEqual(self._test_exit_code(True, True), False)
+        self.assertEqual(self._test_exit_code(False, True), True)
+
+    @staticmethod
+    def _test_exit_code(sim_ok=True, with_has_valid_func=False):

```

Comment:
Call it has_valid_exit_code instead of with_has_valid_func?

Reply:
Related to the next comment, you mean to treat the argument as the variable that is actually returned by the function, instead of a parameter to decide whether to actually redefine the function?

Reply:
Yes that is what I mean. It seems more straightforward to do it that way.